TestGracefulRun: No response when a response was expected

The test suite sporadically fails using v1.2.4 with Go 1.6 on Linux x86_64 with the following output:
```
=== RUN   TestGracefulRun
--- FAIL: TestGracefulRun (1.25s)
        graceful_test.go:44: No response when a response was expected.
=== RUN   TestGracefulRunTimesOut
--- PASS: TestGracefulRunTimesOut (2.60s)
=== RUN   TestGracefulRunDoesntTimeOut
--- PASS: TestGracefulRunDoesntTimeOut (2.00s)
=== RUN   TestGracefulRunNoRequests
--- PASS: TestGracefulRunNoRequests (1.00s)
=== RUN   TestGracefulForwardsConnState
--- PASS: TestGracefulForwardsConnState (2.25s)
=== RUN   TestGracefulExplicitStop
--- PASS: TestGracefulExplicitStop (1.10s)
=== RUN   TestGracefulExplicitStopOverride
--- PASS: TestGracefulExplicitStopOverride (1.10s)
=== RUN   TestBeforeShutdownAndShutdownInitiatedCallbacks
--- PASS: TestBeforeShutdownAndShutdownInitiatedCallbacks (1.10s)
=== RUN   TestNotifyClosed
--- PASS: TestNotifyClosed (0.00s)
=== RUN   TestStopDeadlock
--- PASS: TestStopDeadlock (1.00s)
=== RUN   TestStopRace
--- PASS: TestStopRace (1.00s)
FAIL
exit status 1
FAIL    gopkg.in/tylerb/graceful.v1     14.419s
```

--- a/graceful_test.go
+++ b/graceful_test.go
@@ -82,22 +82,11 @@ func createListener(sleep time.Duration) (*http.Server, net.Listener, error) {
 		rw.WriteHeader(http.StatusOK)
 	})
 
-	time.Sleep(1 * time.Second)
 	server := &http.Server{Addr: ":9654", Handler: mux}
 	l, err := net.Listen("tcp", ":9654")
 	if err != nil {
 	}
 	return server, l, err
-}
-
-func runServer(timeout, sleep time.Duration, c chan os.Signal) error {
-	server, l, err := createListener(sleep)
-	if err != nil {
-		return err
-	}
-
-	srv := &Server{Timeout: timeout, Server: server, interrupt: c}
-	return srv.Serve(l)
 }
 
 func launchTestQueries(t *testing.T, wg *sync.WaitGroup, c chan os.Signal) {
@@ -123,11 +112,16 @@ func TestGracefulRun(t *testing.T) {
 	var wg sync.WaitGroup
 	wg.Add(1)
 
+	server, l, err := createListener(killTime / 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	go func() {
-		runServer(killTime, killTime/2, c)
+		srv := &Server{Timeout: killTime, Server: server, interrupt: c}
+		srv.Serve(l)
 		wg.Done()
 	}()
-	time.Sleep(1 * time.Second)
 
 	wg.Add(1)
 	go launchTestQueries(t, &wg, c)
@@ -140,11 +134,16 @@ func TestGracefulRunTimesOut(t *testing.T) {
 	var wg sync.WaitGroup
 	wg.Add(1)
 
+	server, l, err := createListener(killTime * 10)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	go func() {
-		runServer(killTime, killTime*10, c)
+		srv := &Server{Timeout: killTime, Server: server, interrupt: c}
+		srv.Serve(l)
 		wg.Done()
 	}()
-	time.Sleep(2 * time.Second)
 
 	var once sync.Once
 	wg.Add(1)
@@ -171,11 +170,16 @@ func TestGracefulRunDoesntTimeOut(t *testing.T) {
 	var wg sync.WaitGroup
 	wg.Add(1)
 
+	server, l, err := createListener(killTime * 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	go func() {
-		runServer(0, killTime*2, c)
+		srv := &Server{Timeout: 0, Server: server, interrupt: c}
+		srv.Serve(l)
 		wg.Done()
 	}()
-	time.Sleep(1 * time.Second)
 
 	wg.Add(1)
 	go launchTestQueries(t, &wg, c)
@@ -188,11 +192,16 @@ func TestGracefulRunNoRequests(t *testing.T) {
 	var wg sync.WaitGroup
 	wg.Add(1)
 
+	server, l, err := createListener(killTime * 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	go func() {
-		runServer(0, killTime*2, c)
+		srv := &Server{Timeout: 0, Server: server, interrupt: c}
+		srv.Serve(l)
 		wg.Done()
 	}()
-	time.Sleep(1 * time.Second)
 
 	c <- os.Interrupt
 
@@ -220,8 +229,12 @@ func TestGracefulForwardsConnState(t *testing.T) {
 		http.StateClosed: 8,
 	}
 
+	server, l, err := createListener(killTime / 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	go func() {
-		server, l, _ := createListener(killTime / 2)
 		srv := &Server{
 			ConnState: connState,
 			Timeout:   killTime,
@@ -232,7 +245,6 @@ func TestGracefulForwardsConnState(t *testing.T) {
 
 		wg.Done()
 	}()
-	time.Sleep(2 * time.Second)
 
 	wg.Add(1)
 	go launchTestQueries(t, &wg, c)

